src/helpers/utils.ts

```diff
@@ -131,13 +131,15 @@ export async function unsubscribe(email: string, address: string) {
   );
 }
 
-export async function getVerifiedSubscriptions(batchSize = 1000) {
+export async function getVerifiedSubscriptions(subscription: string, batchSize = 1000) {
   let page = 0;
   let results: SqlRow[] = [];
+  const sub = sanitizeSubscriptions(subscription)[0];
 
   while (true) {
     const result = await db.queryAsync(
-      'SELECT email, address FROM subscribers WHERE verified > 0 ORDER BY created LIMIT ? OFFSET ?',
+      'SELECT email, address, subscriptions FROM subscribers WHERE verified > 0 ' +
+        `AND JSON_CONTAINS(subscriptions, '"${sub}"') ORDER BY created LIMIT ? OFFSET ?`,
```

Comment:
I think better to use `?` . though we are already sanitizing `sub`, using `?` makes the code look more clear and secure :)

Reply:
I think so too, but I haven't been able to get it working, because of the JSON escaping. Can you propose a change ?

Reply:
the problem with this approach is that when a user subscribes the first time we leave `subscription` column entry value equal to null and handle that [here](https://github.com/snapshot-labs/envelop/blob/c86363476375d42ae979c71f2a50a6c024457d50/src/helpers/utils.ts#L177). Since we leave the value empty, this query will not select those users (which haven't updated subscription at least once), but it should.

Reply:
Correct, will look into that

Reply:
Fixed, now taking into account `NULL` subscriptions.